### ISSUE TITLE
when ingesting a lot of new data at once, not all metricdefs make it to ES

### ISSUE THREAD

**Issue by Dieterbe** [raintank/raintank-metric#209]
reported in https://github.com/raintank/ops/issues/270#issuecomment-216351316 and also https://github.com/raintank/ops/issues/283#issuecomment-218336271

I suspect the problem is the ES bulkindexer. i remember reading the code once and it didn't handle failures well, and people were saying how bad it was in the issues too.

**Comment by Dieterbe** [comment]
will need to verify if they do make it after 6h

**Comment by Dieterbe** [comment]
not sure if the "a lot all at once" holds. even after 6h hours, still seeing weirdness like:
```
dieter@rt-tsdb-poc:~$ ./nsq_metrics_to_stdout --nsqd-tcp-address localhost:4150 | grep 'stats.production.timers.grafana.grafana-.*-prod.alert-tickqueue.items.mean'
2016/05/13 16:34:45 INF    1 [metrics/stdout739501#ephemeral] (localhost:4150) connecting to nsqd
2016/05/13 16:34:45 [I] connected to nsqd
2016/05/13 16:34:45 [I] starting listener for http/debug on :6060
2016/05/13 16:34:45 [I] listen tcp :6060: bind: address already in use
12574.6639c142a9c819487bb6926e61cbf72e 10 stats.production.timers.grafana.grafana-1-prod.alert-tickqueue.items.mean_90 1463171690 0 []
12574.58924f56477e1979344ca99377bccc30 10 stats.production.timers.grafana.grafana-1-prod.alert-tickqueue.items.mean_75 1463171690 0 []
12574.d63bb34fb13d86524c4cba588b79631e 10 stats.production.timers.grafana.grafana-1-prod.alert-tickqueue.items.mean 1463171690 0 []
12574.b13e03979ba069b6de773314c7559234 10 stats.production.timers.grafana.grafana-2-prod.alert-tickqueue.items.mean_90 1463171690 0 []
12574.36c695512d4c3da0005057f1a758c9d9 10 stats.production.timers.grafana.grafana-2-prod.alert-tickqueue.items.mean_75 1463171690 0 []
12574.d6ea0f2f8952c5d34dfa001d8874fec7 10 stats.production.timers.grafana.grafana-2-prod.alert-tickqueue.items.mean 1463171690 0 []
12574.98da4603697e9c4472d365aa4ab7b311 10 stats.production.timers.grafana.grafana-3-prod.alert-tickqueue.items.mean_90 1463171690 0 []
12574.0b58c6c0233a2f3f7c8f7bba14ca1790 10 stats.production.timers.grafana.grafana-3-prod.alert-tickqueue.items.mean_75 1463171690 0 []
12574.f7a26e6e47288ebb6dfd83be53d2355b 10 stats.production.timers.grafana.grafana-3-prod.alert-tickqueue.items.mean 1463171690 0 []
```

but:
```
dieter@rt-tsdb-poc:~$ curl http://localhost:9200/metrictank/_search?q=name:stats.production.timers.grafana.grafana-*-prod.alert-tickqueue.items.mean\&pretty=true
{
  "took" : 152,
  "timed_out" : false,
  "_shards" : {
    "total" : 5,
    "successful" : 5,
    "failed" : 0
  },
  "hits" : {
    "total" : 1,
    "max_score" : 1.0,
    "hits" : [ {
      "_index" : "metrictank",
      "_type" : "metric_index",
      "_id" : "12574.f7a26e6e47288ebb6dfd83be53d2355b",
      "_score" : 1.0,
      "_source" : {
        "id" : "12574.f7a26e6e47288ebb6dfd83be53d2355b",
        "org_id" : 12574,
        "name" : "stats.production.timers.grafana.grafana-3-prod.alert-tickqueue.items.mean",
        "metric" : "",
        "interval" : 10,
        "unit" : "",
        "target_type" : "gauge",
        "tags" : [ ],
        "lastUpdate" : 1463134600,
        "nodes" : {
          "n0" : "stats",
          "n1" : "production",
          "n2" : "timers",
          "n3" : "grafana",
          "n4" : "grafana-3-prod",
          "n5" : "alert-tickqueue",
          "n6" : "items",
          "n7" : "mean"
        },
        "node_count" : 8
      }
    } ]
  }
}
```

**Comment by Dieterbe** [comment]
by now all 3 alert-tickqueue.items.mean made it in ES.
but now seeing a similar thing with another metric:

```
dieter@rt-tsdb-poc:~$ ./nsq_metrics_to_stdout --nsqd-tcp-address localhost:4150 | grep 'stats.production.gauges.grafana.grafana.*prod.alert-tickqueue.size'
2016/05/16 04:48:15 INF    1 [metrics/stdout307574#ephemeral] (localhost:4150) connecting to nsqd
2016/05/16 04:48:15 [I] connected to nsqd
2016/05/16 04:48:15 [I] starting listener for http/debug on :6060
2016/05/16 04:48:15 [I] listen tcp :6060: bind: address already in use
12574.275c4747a4964a174df2eb46e0ff166c 10 stats.production.gauges.grafana.grafana-2-prod.alert-tickqueue.size 1463388500 20 []
12574.adce8105af4fc5ce392fa6d54a71b1ff 10 stats.production.gauges.grafana.grafana-3-prod.alert-tickqueue.size 1463388500 20 []
12574.891ffea38a2fbe0264b40c66a487ec6c 10 stats.production.gauges.grafana.grafana-1-prod.alert-tickqueue.size 1463388500 20 []
12574.891ffea38a2fbe0264b40c66a487ec6c 10 stats.production.gauges.grafana.grafana-1-prod.alert-tickqueue.size 1463388510 20 []
12574.275c4747a4964a174df2eb46e0ff166c 10 stats.production.gauges.grafana.grafana-2-prod.alert-tickqueue.size 1463388510 20 []
12574.adce8105af4fc5ce392fa6d54a71b1ff 10 stats.production.gauges.grafana.grafana-3-prod.alert-tickqueue.size 1463388510 20 []
12574.275c4747a4964a174df2eb46e0ff166c 10 stats.production.gauges.grafana.grafana-2-prod.alert-tickqueue.size 1463388520 20 []
12574.adce8105af4fc5ce392fa6d54a71b1ff 10 stats.production.gauges.grafana.grafana-3-prod.alert-tickqueue.size 1463388520 20 []
12574.891ffea38a2fbe0264b40c66a487ec6c 10 stats.production.gauges.grafana.grafana-1-prod.alert-tickqueue.size 1463388520 20 []
12574.adce8105af4fc5ce392fa6d54a71b1ff 10 stats.production.gauges.grafana.grafana-3-prod.alert-tickqueue.size 1463388530 20 []
12574.891ffea38a2fbe0264b40c66a487ec6c 10 stats.production.gauges.grafana.grafana-1-prod.alert-tickqueue.size 1463388530 20 []
```

ES only lists this for 1 and 3, not 2, whereas index.json lists 1, 2 and 3. i did the ES test before and after. so since the built-in index has it i now suspect the ES indexing even more.

```
dieter@rt-tsdb-poc:~$ curl -H 'x-org-id:12574' http://localhost:18763/metrics/index.json | tr , '\n' | grep alert-tickqueue.size
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0"stats.development.gauges.grafana.standalone-1.alert-tickqueue.size"
"stats.production.gauges.grafana.grafana-1-prod.alert-tickqueue.size"
"stats.production.gauges.grafana.grafana-2-prod.alert-tickqueue.size"
"stats.production.gauges.grafana.grafana-3-prod.alert-tickqueue.size"
100 3629k    0 3629k    0     0  16.8M      0 --:--:-- --:--:-- --:--:-- 16.9M

```

**Comment by Dieterbe** [comment]
fixed by #213
